agr_java_core/src/main/java/org/alliancegenome/core/translators/document/PhenotypeTranslator.java

```diff
@@ -0,0 +1,281 @@
+                })
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    private Map<Gene, List<PhenotypeEntityJoin>> getGeneAnnotationMap(Phenotype phenotype, Gene gene) {
+        // group by gene then by association type
+        Map<Gene, List<PhenotypeEntityJoin>> geneMap = phenotype.getPhenotypeEntityJoins().stream()
+                .filter(phenotypeEntityJoin -> gene == null || phenotypeEntityJoin.getGene().equals(gene))
+                .collect(groupingBy(PhenotypeEntityJoin::getGene));
+
+        // sort by gene symbol
+        return geneMap.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    public Map<Gene, Map<Optional<Feature>, List<PhenotypeEntityJoin>>> getGeneFeatureAnnotationMap(Phenotype phenotype, Gene gene) {
+        // group by gene then by feature
+        Map<Gene, Map<Optional<Feature>, List<PhenotypeEntityJoin>>> geneMap = phenotype.getPhenotypeEntityJoins().stream()
+                .filter(phenotypeEntityJoin -> gene == null || phenotypeEntityJoin.getGene().equals(gene))
+                .collect(
+                        groupingBy(PhenotypeEntityJoin::getGene,
+                                groupingBy(join -> Optional.ofNullable(join.getFeature())))
+                );
+
+        // sort by gene symbol
+        return geneMap.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private PublicationDoclet getPublicationDoclet(PhenotypeEntityJoin association, Publication publication) {
+        PublicationDoclet pubDoc = new PublicationDoclet();
+        pubDoc.setPrimaryKey(publication.getPrimaryKey());
+
+        pubDoc.setPubMedId(publication.getPubMedId());
+        pubDoc.setPubMedUrl(publication.getPubMedUrl());
+        pubDoc.setPubModId(publication.getPubModId());
+        pubDoc.setPubModUrl(publication.getPubModUrl());
+
+/*
+        Set<String> evidencesDocument = association.getEvidenceCodes().stream()
+                .map(EvidenceCode::getPrimaryKey)
+                .collect(Collectors.toSet());
+        pubDoc.setEvidenceCodes(evidencesDocument);
+*/
+        return pubDoc;
+    }
+
+    private PhenotypeDocument getPhenotypeDocument(Phenotype phenotype) {
+        return getPhenotypeDocument(phenotype, false);
+    }
+
+
+    private PhenotypeDocument getPhenotypeDocument(Phenotype phenotype, boolean shallow) {
+        PhenotypeDocument document = new PhenotypeDocument();
+        document.setPrimaryKey(phenotype.getPrimaryKey());
+        document.setPrimaryId(phenotype.getPrimaryKey());
+        document.setPhenotypeStatement(phenotype.getPhenotypeStatement());
+        if (shallow)
+            return document;
+
+        return document;
+    }
+
+    private List<SourceDoclet> getSourceUrls(DOTerm doTerm) {
+
```

Comment:
This code is duplicate from the DiseaseTranslator.java shouldn't this be put into one place?

Reply:
This method is not used in the PhenotypeTranslator. Removing it altogether.